client_push_pre_go17.go

```diff
@@ -0,0 +1,53 @@
+// +build !go1.7
+
+package apns2
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"golang.org/x/net/context"
+	"golang.org/x/net/context/ctxhttp"
+)
+
+// Push sends a Notification to the APNs gateway. If the underlying http.Client
+// is not currently connected, this method will attempt to reconnect
+// transparently before sending the notification. It will return a Response
+// indicating whether the notification was accepted or rejected by the APNs
+// gateway, or an error if something goes wrong.
+//
+// Context with deadline allows to close long request when context timeout exceed. It can be nil, for back compatibility
+func (c *Client) PushWithCtx(n *Notification, ctx context.Context) (*Response, error) {
```

Comment:
There should be a way to minimize the duplication between the Go 1.6 and Go 1.7+ versions of this.

`context.Context` is an interface with the same definition in both packages, so they are interchangeable. 

https://golang.org/pkg/context/#Context
https://godoc.org/golang.org/x/net/context#Context

I'm not sure the best way to do this, but even redefining the `Context` interface locally should allow either one to be passed in.